### DISCUSSION TITLE
Gradient accumulation + DeepSpeed LR scheduler

### DISCUSSION THREAD

**Discussion by gahdritz** [Lightning-AI/pytorch-lightning#11686] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
How does gradient accumulation interact with DeepSpeed learning rate scheduling (e.g. the per-step warm-up scheduler)? Is the learning rate updated after every iteration, or only after the model weights are ultimately updated?

**Comment by rohitgr7** [comment, marked as answer]
it considers the accumulation before doing lr_scheduler_step:
https://github.com/PyTorchLightning/pytorch-lightning/blob/86b177ebe5427725b35fde1a8808a7b59b8a277a/pytorch_lightning/loops/epoch/training_epoch_loop.py#L387-L390
